arch/arm/chunkset_neon.c

```diff
@@ -23,11 +23,15 @@ static inline void chunkmemset_1(uint8_t *from, chunk_t *chunk) {
 }
 
 static inline void chunkmemset_2(uint8_t *from, chunk_t *chunk) {
-    *chunk = vreinterpretq_u8_s16(vdupq_n_s16(*(int16_t *)from));
+    int16_t tmp;
```

Comment:
Consider this instead:
```
int16_t tmp;
#if defined(UNALIGNED_OK)
    tmp = *(int16_t*)from;
#else
    tmp = (int16_t)from[0] << 8 | from[1]; // ? not sure of the order here
#endif
*chunk = vreinterpretq_u8_s16(vdupq_n_s16(tmp));
```

Reply:
`UNALIGNED_OK` is wrong test as it is used for copies between memory and general-purpose registers, not copies between memory and vector registers... NEON is not the only vector instruction set that causes exception with unaligned loads and stores wider than 8 bits, even SSE has restriction even though it has unaligned version for some instructions.

Reply:
It is still unaligned read/load regardless of source/destination right? I think `UNALIGNED_OK` should cover both unaligned support for intrinsic and also architecture unaligned support.

Reply:
@nmoinvaz That would be seriously bad idea... For the same reason we introduced separate define for 64-bit integers.

Reply:
I don't get it but it's ok.

Reply:
@nmoinvaz 

1. Disabling `UNALIGNED_OK` would unnecessarily slow down irrelevant parts of the code
2. It would redefine purpose of `UNALIGNED_OK` the second time
3. Adding another define for just one part of the code is premature
4. Point of using memcpy is to forbid the code optimizer to optimize away the temporary variable as "register" keyword is now deprecated.

Reply:
Good explanation I got it now thank you. My only suggestion then is to put in the commit message that `memcpy` is used to prevent temporary variable from being optimized way.

Reply:
@nmoinvaz Using memcpy() to alias pointer types of different bit width has been used elsewhere already... 

It usually works better than shifts due to some platforms being both big and little endian or using floating point registers for integer vector elements (vector registers are by default in floating point mode). 

Optimizer of gcc is also smart enough to optimize away redundant byte swaps using shifts.